### ISSUE TITLE
Recent changes breaking CI

### ISSUE THREAD

**Issue by oddhack** [KhronosGroup/Vulkan-Hpp#578]
CI jobs in vulkan internal gitlab that were working 18 hours ago are now getting a lot of warnings like this:

In file included from tests/hpptest.cpp:1:
325 Vulkan-Hpp/vulkan/vulkan.hpp:6158:81: note: declared here
326    struct [[deprecated("vk::cpp_type is deprecated. Use vk::CppType instead.")]] cpp_type
cc1plus: all warnings being treated as errors

I re-ran previously successful jobs to confirm and they fail in this fashion. This isn't actually hard-failing CI since we have the hpp-generate / hpp-compile stages marked as optional, but it's going to generate warnings on every job until it's fixed.

**Comment by asuessenbach** [comment]
#579 did not fix this issue. Reopening.

**Comment by wumo** [comment]
I tested with [Vulkan-Headers v1.2.142](https://github.com/KhronosGroup/Vulkan-Headers/tree/v1.2.142). It still print lots of warnings of 
```
warning: ‘template<vk::ObjectType value> struct vk::cpp_type’ is deprecated: vk::cpp_type is deprecated. Use vk::CppType instead. [-Wdeprecated-declarations]
```

**Comment by asuessenbach** [comment]
@wumo, could you please check if those warnings are still there?
If so, what compiler are you using?

**Comment by wumo** [comment]
@asuessenbach  Those warnings still exists on `gcc-10`, which is :
```
gcc-10 (Ubuntu 10-20200411-0ubuntu1) 10.0.1 20200411 (experimental) [master revision bb87d5cc77d:75961caccb7:f883c46b4877f637e0fa5025b4d6b5c9040ec566]
```

**Comment by asuessenbach** [comment]
Resolved by #631.
